packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -570,6 +570,47 @@ contract OptimismPortal_Test is CommonTest {
         // Check that the balance has been correctly updated
         assertEq(optimismPortal.balance(), address(optimismPortal).balance);
     }
+
+    /// @dev Tests that the donateETH function donates ETH and does no state read/write
+    function test_donateETH_succeeds(uint256 _amount) external {
+        vm.startPrank(alice);
+        vm.deal(alice, _amount);
+
+        uint256 preBalance = address(optimismPortal).balance;
+
+        vm.startStateDiffRecording();
+        optimismPortal.donateETH{ value: _amount }();
+        VmSafe.AccountAccess[] memory accountAccesses = vm.stopAndReturnStateDiff();
+
+        // not necessary since it's checked below
+        assertEq(address(optimismPortal).balance, preBalance + _amount);
+
+        // 0 for extcodesize of proxy before being called by this test,
+        // 1 for the call to the proxy by the pranked address
+        // 2 for the delegate call to the impl by the proxy
+        assertEq(accountAccesses.length, 3);
```

Comment:
We can probably do this without the state diff recording, right? 

Really just need to check that the balance increased

Reply:
yeah, maybe, added it as an extra check to ensure nothing happens (read or write) regarding storage in the contract

Reply:
should it be removed then?

Reply:
Honestly it's fine, this seems reasonable as-is.